File: typescript/leetCode/dp/ClimbingStairs.ts
export function climbStairs(n: number): number {
    if(n === 0) return 0;
    if(n === 1) return 1;
    if(n === 2) return 2;
    let tmp1 : number = 1;
    let tmp2 : number = 2;
    let answer : number = 0;
    for(let i : number = 2; i < n; i++){
        answer = tmp1 + tmp2;
        tmp1 = tmp2;
        tmp2 = answer;
    }
    console.log(answer);
    return answer;
};

File: rust/codingtest/src/leet_code/dp/climbing_stairs.rs
struct Solution;

impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        if n == 0 { return 0;}
        else if n == 1 { return 1; }
        else if n == 2 { return 2; }

        let mut tmp1 = 1;
        let mut tmp2 = 2;
        let mut answer = 0;

        for i in 2..n{
            answer = tmp1 + tmp2;
            tmp1 = tmp2;
            tmp2 = answer;
        }
        answer
    }
}

pub fn run(){
    let n = Solution::climb_stairs(5);
    println!("{}", n);
}
// 0 => 0
// 1 => 1
// 2 => 2
// 3 => 3
// 4 => 5
// 5 => 8
// 6 => 13
// 7 => 21
// 8 => 34
// 9 => 55